tooling/noir_js/test/node/e2e.test.ts

```diff
@@ -6,148 +6,118 @@ import { CompiledCircuit } from '@noir-lang/types';
-  const proof = await program.generateFinalProof(inputs);
-
-  // Proof verification
-  const isValid = await program.verifyFinalProof(proof);
-  expect(isValid).to.be.true;
+it('end-to-end proof creation and verification (outer)', async () => {
+  before(async () => {
+    backend = new Backend(assert_lt_program, { numOfThreads: 4 });
+    noir = new Noir(backend);
+  });
+
+  it('Creates and verifies end-to-end outer proofs with underlying backend API', async () => {
+    // Noir.Js part
+    const { witness } = await noir.execute(inputs);
+
+    // BackendBarretenberg part
+    const prover = new Backend(assert_lt_program, { numOfThreads: 4 });
+    const proof = await prover.generateFinalProof(witness);
+    const isValid = await prover.verifyFinalProof(proof);
+
+    // tests
+    expect(isValid).to.be.true;
+  });
+
+  it('Creates and verifies end-to-end outer proofs with Noir API', async () => {
+    const proof = await noir.generateFinalProof(inputs);
+    const isValid = await noir.verifyFinalProof(proof);
+
+    // tests
+    expect(isValid).to.be.true;
+  });
 });
 
-it('end-to-end proof creation and verification (inner)', async () => {
-  // Noir.Js part
-  const inputs = {
-    x: '2',
-    y: '3',
-  };
-
-  const program = new Noir(assert_lt_program);
-
-  const { witness } = await program.execute(inputs);
-
-  // bb.js part
+describe('Inner proofs', () => {
+  before(() => {
+    backend = new Backend(assert_lt_program, { numOfThreads: 4 });
+    noir = new Noir(); // backendless noir;
+  });
+
+  it('Creates and verifies end-to-end inner proofs with underlying backend API', async () => {
+    // Noir.Js part
+    const { witness } = await noir.execute(inputs, assert_lt_program); // backendless noir, so it needs a circuit
+    const proof = await backend.generateIntermediateProof(witness);
+    const isValid = await backend.verifyIntermediateProof(proof);
+    expect(isValid).to.be.true;
+  });
+
+  // The "real" workflow will involve a prover and a verifier on different systems.
   //
-  // Proof creation
-  const prover = new Backend(assert_lt_program);
-  const proof = await prover.generateIntermediateProof(witness);
-
-  // Proof verification
-  const isValid = await prover.verifyIntermediateProof(proof);
-  expect(isValid).to.be.true;
-});
-
-// The "real" workflow will involve a prover and a verifier on different systems.
-//
-// We cannot do this in our tests because they will panic with:
-// `RuntimeError: null function or function signature mismatch`
-//
-// This happens when we we create a proof with one barretenberg instance and
-// try to verify it with another.
-//
-// If this bug is fixed, we can remove this test and split barretenberg into
-// a prover and verifier class to more accurately reflect what happens in production.
-//
-// If its not fixable, we can leave it in as documentation of this behavior.
-it('[BUG] -- bb.js null function or function signature mismatch (different instance) ', async () => {
```

Comment:
Can you explain what's going on with these deletions? I'm not clear on how these connect with the rest of the PR.

Reply:
yeah github is fucking up, file is fine